test/controllers/previousRegistrations/PreviouslyRegisteredControllerSpec.scala

```diff
@@ -33,83 +36,112 @@ import views.html.previousRegistrations.PreviouslyRegisteredView
 
-        val result = route(application, request).value
+        forAll(allModeWaypoints) { case (_, waypoints) =>
 
-        val view = application.injector.instanceOf[PreviouslyRegisteredView]
+          val request = FakeRequest(GET, previouslyRegisteredRoute(waypoints))
+
+          val result = route(application, request).value
+
+          val view = application.injector.instanceOf[PreviouslyRegisteredView]
+
+          status(result) mustEqual OK
+          contentAsString(result) mustEqual view(form, waypoints)(request, messages(application)).toString
 
-        status(result) mustEqual OK
-        contentAsString(result) mustEqual view(form, waypoints)(request, messages(application)).toString
+        }
       }
     }
 
-    "must populate the view correctly on a GET when the question has previously been answered" in {
-
+    "must populate the view correctly on a GET when the question has previously been answered as true when not in Amend mode" in {
       val userAnswers = UserAnswers(userAnswersId).set(PreviouslyRegisteredPage, true).success.value
-
       val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
 
       running(application) {
-        val request = FakeRequest(GET, previouslyRegisteredRoute)
+        forAll(nonAmendModeWaypoints) { case (_, waypoints) =>
+          val request = FakeRequest(GET, previouslyRegisteredRoute(waypoints))
 
-        val view = application.injector.instanceOf[PreviouslyRegisteredView]
+          val view = application.injector.instanceOf[PreviouslyRegisteredView]
 
-        val result = route(application, request).value
+          val result = route(application, request).value
 
-        status(result) mustEqual OK
-        contentAsString(result) mustEqual view(form.fill(true), waypoints)(request, messages(application)).toString
+          status(result) mustEqual OK
+          contentAsString(result) mustEqual view(form.fill(true), waypoints)(request, messages(application)).toString
+        }
       }
     }
 
-    "must save the answer and redirect to the next page when valid data is submitted" in {
+    "must fail on GET when the existing answer is true when in Amend mode" in {
+      val userAnswers = UserAnswers(userAnswersId).set(PreviouslyRegisteredPage, true).success.value
+      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, previouslyRegisteredRoute(amendModeWaypoints))
+        val result = route(application, request).value
 
-      val mockSessionRepository = mock[AuthenticatedUserAnswersRepository]
+        result.failed.futureValue mustBe an[InvalidAmendModeOperationException]
+      }
+    }
 
-      when(mockSessionRepository.set(any())) thenReturn Future.successful(true)
+    "must save the answer and redirect to the next page when valid data is submitted when the answer is originally true" in {
```

Comment:
In the next step we will guard against amend mode at this point as we are going to do existing registration comparisons which we have not started yet